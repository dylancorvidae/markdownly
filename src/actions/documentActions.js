export const UPDATE_MARKDOWN = 'UPDATE_MARKDOWN';
export const updateMarkdown = updatedInput => ({
  type: UPDATE_MARKDOWN,
  payload: updatedInput
});

export const ADD_TAB = 'ADD_TAB';
export const addTab = title => ({
  type: ADD_TAB,
  payload: title
});

export const DELETE_TAB = 'DELETE_TAB';
export const deleteTab = title => ({
  type: DELETE_TAB,
  payload: title
});

export const ADD_SEARCH = 'ADD_SEARCH';
export const addSearch = searchByTitle => ({
  type: ADD_SEARCH,
  payload: searchByTitle
});

export const GET_MATCH_TITLE = 'GET_MATCH_TITLE';
export const getMatchTitle = searchTerm => ({
  type: GET_MATCH_TITLE,
  payload: searchTerm
});

