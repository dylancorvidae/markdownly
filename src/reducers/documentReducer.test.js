import React from 'react';
import { shallow } from 'enzyme';
import reducer from '../reducers/index';
import { updateMarkdown, addTab } from '../actions/documentActions';

describe('documentReducer component', () => {
  it('renders documentReducer', () => {
    const wrapper = shallow(<documentReducer />);
    expect(wrapper).toMatchSnapshot();
  });

  it('handles unknown case', () => {
    const newState = reducer(undefined, { type: 'thing' });
    expect(newState.documentReducer).toEqual({
      active: 'default',
      markdowns: {
        default: '# hi there',
      },
      searchTerm: ''
    });
  });

  it('handles update markdown case', () => {
    const state = {
      active: 'default',
      markdowns: {
        default: '# hi there',
      },
      searchTerm: ''
    };

    const action = updateMarkdown('yo whats up');
    const newState = reducer(state.documentReducer, action);

    expect(newState.documentReducer).toEqual({
      ...state,
      markdowns: { ...state.markdowns, default: 'yo whats up' }
    });
  });

  it('handles adding a tab case', () => {
    const state = {
      active: 'default',
      markdowns: {
        default: '# hi there',
      },
      searchTerm: ''
    };
   
    const action = addTab('1');
    const newState = reducer(state.documentReducer, action);

    expect(newState.documentReducer).toEqual({
      ...state,
      markdowns: { ...state.markdowns, 1: '# 1' }
    });
  });
});
