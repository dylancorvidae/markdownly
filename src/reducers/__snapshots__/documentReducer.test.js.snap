// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`documentReducer component renders documentReducer 1`] = `<documentReducer />`;
